rest_framework_json_api/settings.py

```diff
@@ -27,6 +32,22 @@ def __init__(self, user_settings=settings, defaults=DEFAULTS):
         self.defaults = defaults
         self.user_settings = user_settings
 
+        value = getattr(
+            self.user_settings,
+            JSON_API_SETTINGS_PREFIX + 'NESTED_SERIALIZERS_RENDERING_STRATEGY',
+            self.defaults['NESTED_SERIALIZERS_RENDERING_STRATEGY'])
+
+        if value not in (RELATIONS_RENDERING_STRATEGY, ATTRIBUTE_RENDERING_STRATEGY):
+            raise AttributeError("Invalid value '%s' for JSON API setting "
+                                 "NESTED_SERIALIZERS_RENDERING_STRATEGY" % value)
+        if value == RELATIONS_RENDERING_STRATEGY and \
+                not hasattr(self.user_settings,
+                            JSON_API_SETTINGS_PREFIX + 'NESTED_SERIALIZERS_RENDERING_STRATEGY'):
+            warnings.warn(DeprecationWarning(
```

Comment:
The nested serializer issue currently doesn't affect many users but with this deprecation warning every user will be notified about a case which might not concern them at all.

I also think a deprecation warning should be more specific which states what serializer and what field is affected. Otherwise it is really for a DJA user hard to figure out what needs to be changed. So I suggest to move this deprecation warning into the renderer.